src/main/java/org/apache/maven/shared/filtering/DefaultMavenResourcesFiltering.java

```diff
@@ -90,23 +87,25 @@ public boolean filteredFileExtension( String fileName, List<String> userNonFilte
         {
             nonFilteredFileExtensions.addAll( userNonFilteredFileExtensions );
         }
-        String extension = StringUtils.lowerCase( FilenameUtils.getExtension( fileName ) );
+        String extension = getExtension( fileName );
         boolean filteredFileExtension = !nonFilteredFileExtensions.contains( extension );
-        if ( getLogger().isDebugEnabled() )
+        if ( LOGGER.isDebugEnabled() )
         {
-            getLogger().debug( "file " + fileName + " has a" + ( filteredFileExtension ? " " : " non " )
+            LOGGER.debug( "file " + fileName + " has a" + ( filteredFileExtension ? " " : " non " )
                 + "filtered file extension" );
         }
         return filteredFileExtension;
     }
 
+    private static String getExtension( String fileName )
+    {
+        String rawExt = FilenameUtils.getExtension( fileName );
+        return rawExt == null ? null : rawExt.toLowerCase( Locale.ROOT );
+    }
+
     @Override
     public List<String> getDefaultNonFilteredFileExtensions()
     {
-        if ( this.defaultNonFilteredFileExtensions == null )
-        {
-            this.defaultNonFilteredFileExtensions = new ArrayList<>();
-        }
         return this.defaultNonFilteredFileExtensions;
```

Comment:
This changes semantics, no?

Reply:
Nope, member made final, no more "lazy init"

Reply:
Dammit, not noticed.